crates/shared/src/bad_token/cache.rs

```diff
@@ -94,11 +86,12 @@ impl CachingDetector {
                 let start = Instant::now();
```

Comment:
Looks like this loop can be written without a bunch of allocations and an extra scope.

<details>
<summary>code</summary>

```rust
            loop {
                let start = Instant::now();

                let futures = detector.cache.iter().filter_map(|entry| {
                    let (token, (instant, _)) = entry.pair();
                    let (token, instant) = (*token, *instant);
                    if start.saturating_duration_since(instant) < prefetch_time_to_expire {
                        return None;
                    }
                    let detector = detector.clone();
                    Some(async move {
                        match detector.inner.detect(token).await {
                            Ok(result) => Some((token, result)),
                            Err(err) => {
                                tracing::warn!(
                                    ?token,
                                    ?err,
                                    "unable to determine token quality in the background task"
                                );
                                None
                            }
                        }
                    })
                });

                let results = join_all(futures).await;
                detector.insert_many_into_cache(results.into_iter().flatten());

                let remaining_sleep = maintenance_timeout.saturating_sub(start.elapsed());
                tokio::time::sleep(remaining_sleep).await;
            }
```
</details>